src/main/kotlin/app/revanced/patches/youtube/layout/hide/getpremium/bytecode/patch/HideGetPremiumVideoAdvertisementPatch.kt

```diff
@@ -0,0 +1,63 @@
+package app.revanced.patches.youtube.layout.hide.getpremium.bytecode.patch
+
+import app.revanced.extensions.toErrorResult
+import app.revanced.patcher.annotation.Description
+import app.revanced.patcher.annotation.Name
+import app.revanced.patcher.annotation.Version
+import app.revanced.patcher.data.BytecodeContext
+import app.revanced.patcher.extensions.addInstructions
+import app.revanced.patcher.extensions.instruction
+import app.revanced.patcher.patch.BytecodePatch
+import app.revanced.patcher.patch.PatchResult
+import app.revanced.patcher.patch.PatchResultSuccess
+import app.revanced.patcher.patch.annotations.DependsOn
+import app.revanced.patcher.patch.annotations.Patch
+import app.revanced.patches.youtube.layout.hide.getpremium.annotations.HideGetPremiumVideoAdvertisementCompatibility
+import app.revanced.patches.youtube.layout.hide.getpremium.bytecode.fingerprints.GetPremiumVideoAdvertisementViewFingerprint
+import app.revanced.patches.youtube.layout.hide.getpremium.resource.patch.HideGetPremiumVideoAdvertisementResourcePatch
+import app.revanced.patches.youtube.misc.integrations.patch.IntegrationsPatch
+import org.jf.dexlib2.iface.instruction.TwoRegisterInstruction
+
+@Patch
+@DependsOn([IntegrationsPatch::class, HideGetPremiumVideoAdvertisementResourcePatch::class])
+@Name("hide-get-premium")
+@Description("Hides advertisements for youtube premium in the video player.")
+@HideGetPremiumVideoAdvertisementCompatibility
+@Version("0.0.1")
+class HideGetPremiumVideoAdvertisementPatch : BytecodePatch(
+    listOf(
+        GetPremiumVideoAdvertisementViewFingerprint,
+    )
+) {
+    override fun execute(context: BytecodeContext): PatchResult {
+        GetPremiumVideoAdvertisementViewFingerprint.result?.let {
+            it.mutableMethod.apply {
+                val startIndex = it.scanResult.patternScanResult!!.startIndex
+                val measuredWidthRegister = (instruction(startIndex) as TwoRegisterInstruction).registerA
+                val measuredHeightInstruction = instruction(startIndex + 1) as TwoRegisterInstruction
+                val measuredHeightRegister = measuredHeightInstruction.registerA
+                val tempRegister = measuredHeightInstruction.registerB
+
+                addInstructions(
+                    startIndex + 2,
+                    """
+                        invoke-static {}, $INTEGRATIONS_CLASS_DESCRIPTOR->hideGetPremiumView()Z
+                        move-result v$tempRegister
+                        if-eqz v$tempRegister, :allow
+                        const/4 v$measuredWidthRegister, 0x0
+                        const/4 v$measuredHeightRegister, 0x0
+                        :allow
+                        nop
```

Comment:
This could possibly be refactored. It is generally better to outsource as much logic from patches to integrations as possible so that it works reliably. Therefore, the method `hideGetPremiumView` could accept a `View` object and set its visibility. This would get rid of the logic in the patch. There are a couple of similar patches that can be taken as an example.

Reply:
Seems like you have tried something of that sort already [here](https://github.com/revanced/revanced-patches/pull/2020#issuecomment-1528802269). Have you tried setting the visibility of the view?